packages/cli/src/commands/export-dynamic-plugin/backend.ts

```diff
@@ -0,0 +1,320 @@
+          `Embeded package '${pkgToEmbed}' is not part of direct dependencies.Are you sure you want to embed it ?`,
+        );
+      }
+      mergeWithOutput.push(pkgToEmbed);
+      const relatedCommonPackage = pkgToEmbed.replace(/-backend$/, '-common');
+      if (relatedCommonPackage !== pkgToEmbed) {
+        mergeWithOutput.push(relatedCommonPackage);
+      }
+    }
+  }
+
+  const filter = {
+    include: mergeWithOutput,
+    exclude: mergeWithOutput.length !== 0 ? undefined : /.*/,
+  };
+  const moveToPeerDependencies: (string | RegExp)[] = [/@backstage\//];
+
+  if (opts.sharedPackage !== undefined) {
+    for (const sharedPkg of opts.sharedPackage as string[]) {
+      moveToPeerDependencies.push(sharedPkg);
+    }
+  }
+
+  const rollupConfigs = await makeRollupConfigs({
+    outputs,
+    minify: Boolean(opts.minify),
+    useApiExtractor: false,
+  });
+
+  if (rollupConfigs.length === 0) {
+    throw new Error('Rollup config is missing');
+  }
+
+  const dependenciesToAdd: {
+    [key: string]: string;
+  } = {};
+
+  const rollupConfig = rollupConfigs[0];
+  rollupConfig.plugins?.push(
+    embedModules({
+      filter: filter,
+      addDependency(embeddedModule, dependencyName, dependencyVersion) {
+        const existingVersion = dependenciesToAdd[dependencyName];
+        if (existingVersion === undefined) {
+          dependenciesToAdd[dependencyName] = dependencyVersion;
+          return;
+        }
+        if (existingVersion !== dependencyVersion) {
+          throw new Error(
+            `several versions ('${existingVersion}', '${dependencyVersion}') of the same transitive dependency ('${dependencyName}') for embedded module ('${embeddedModule}')`,
+          );
+        }
+      },
+    }),
+  );
+
+  if (Array.isArray(rollupConfig.output)) {
+    rollupConfig.output.forEach(output => {
+      if (output.format === 'commonjs') {
+        output.interop = 'default';
+      }
+    });
+  }
+
+  await fs.remove(paths.resolveTarget('dist'));
+
+  try {
+    const bundle = await rollup(rollupConfig);
+    if (rollupConfig.output) {
+      for (const output of [rollupConfig.output].flat()) {
+        await bundle.generate(output);
+        await bundle.write(output);
+      }
+    }
+  } catch (error) {
+    throw new Error(formatErrorMessage(error));
+  }
+
+  if (opts.clean) {
+    await fs.remove(target);
+  }
+
+  const monoRepoPackages = await getPackages(paths.targetDir);
+  await productionPack({
+    packageDir: '',
+    targetDir: target,
+    customizeManifest: (pkgToCustomize: BackstagePackageJson) => {
+      function test(str: string, expr: string | RegExp): boolean {
+        if (typeof expr === 'string') {
+          return str === expr;
+        }
+        return expr.test(str);
+      }
+
+      pkgToCustomize.name = `${pkgToCustomize.name}-dynamic`;
+      (pkgToCustomize as any).bundleDependencies = true;
+      pkgToCustomize.scripts = undefined;
+      pkgToCustomize.types = undefined;
```

Comment:
Out of curiosity, is there a reason to explicitly remove these?

Reply:
This package has no sources, and no action that is to be done on sources (such as lint, build, etc ...). It is already derived from the source package. I don't see how it makes sense to keep scripts or even types in the corresponding package.json.